DESIGN.md

```diff
@@ -0,0 +1,99 @@
+# Shipit - Opinionated release management for Rust crates
+
+Shipit is intended to aid in the release management process of Rust projects
+including Tokio, Tower, and Mio. While it offers some level of configuration, it
+is opinionated in the process that developers will take. Projects that stick
+with the Shipit process will gain automated publishing to crates.io including
+changelog generation.
+
+## Project workflow with Shipit.
+
+In order to work with Shipit, projects should be contained in a [Git] repository
+and structured to follow [Cargo] idioms. Shipit supports both single crates and
+[Cargo workspaces][workspace] containing multiple crates. When the repository
+contains a workspace with multiple crates, additional requirements are described
+below.
+
+Shipit requires that the master branch is always able to be released. This means
+that changes must be fully implemented in branches and merged in PRs. It also
+means that breaking changes should not be applied to the master branch until the
+project maintainers are ready for a breaking change release to be issued.
+
+Shipit can be added to a project at any time. As such, it must be tolerant of
+conventions used by the project before it adopted Shipit. However, it is
+required that the project switches to Shipit conventions at the point of
+adoption.
+
+### Github workflow
+
+Projects **must** be hosted on [Github]. All changes **must** be submited via
+[pull requests][pr]. The [Github] repository **should** be configured to protect
+the master branch, preventing commits to be pushed directly to the master
+branch.
+
+#### Labels
+
+Shipit requires each pull request to be labeled with one of the following:
```

Comment:
Anyway, these could be configurable? Say set it to `type: bug` from just `bug`?

Reply:
They should be configurable... I wasn't sure if specifying that was worth it at this level. Probably easy enough to have a note.

Reply:
Yeah, no need to define it just maybe update the language to indicate it will be possible 😄